modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java

```diff
@@ -2318,6 +2318,66 @@ public void testEnableAccountRestricted() throws Exception {
 					ObjectRelationshipConstants.TYPE_ONE_TO_MANY, null)));
 	}
 
+	@Test
+	public void testPublishCompanyObjectDefinition() throws Exception {
+		ObjectDefinition objectDefinition1 =
+			ObjectDefinitionTestUtil.publishObjectDefinition();
+
+		PermissionChecker originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+		String originalName = PrincipalThreadLocal.getName();
+
+		Company company = CompanyTestUtil.addCompany();
+
+		PortalInstances.initCompany(company);
+
+		try (SafeCloseable safeCloseable =
+				CompanyThreadLocal.setCompanyIdWithSafeCloseable(
+					company.getCompanyId())) {
+
+			User user = UserTestUtil.getAdminUser(company.getCompanyId());
+
+			PermissionThreadLocal.setPermissionChecker(
+				PermissionCheckerFactoryUtil.create(user));
+			PrincipalThreadLocal.setName(user.getUserId());
+
+			ObjectDefinition objectDefinition2 =
+				_objectDefinitionLocalService.addCustomObjectDefinition(
+					user.getUserId(), 0, objectDefinition1.getClassName(),
+					false, false, true, false, false, false, false, null,
+					LocalizedMapUtil.getLocalizedMap(
+						objectDefinition1.getLabel()),
+					objectDefinition1.getShortName(), null, null,
+					LocalizedMapUtil.getLocalizedMap(
+						objectDefinition1.getPluralLabel()),
+					true, ObjectDefinitionConstants.SCOPE_COMPANY,
+					ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT,
+					Collections.emptyList(),
+					Arrays.asList(
+						ObjectFieldUtil.createObjectField(
+							ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+							ObjectFieldConstants.DB_TYPE_STRING,
+							RandomTestUtil.randomString(),
+							StringUtil.randomId())));
```

Comment:
@victorKammerer use `TextObjectFieldBuilder` because we are gonna replace all usages of `ObjectFieldUtil` with `ObjectFieldBuilder`